account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,456 @@
+                name=self.name,
+                api_name=self.api_name
+            )
+        )
+
+        if token_record.access_token and token_record.refresh_token:
+            self._logger.info(
+                "Connection button Clicked - endpoint name {name}, ".format(name=self.name) +
+                "We have access token and refresh token"
+            )
+            version = self._json_command('version')
+            return version
+        else:
+            self._logger.info(
+                "Connection button Clicked - endpoint name {name}, No access token ".format(name=self.name) +
+                "found and no refresh_token found from the token record table."
+            )
+            authorisation_tracker = self.env['mtd.api_request_tracker'].search([('closed', '=', False)])
+            self._logger.info(
+                "Connection button Clicked - endpoint name {name}, ".format(name=self.name) +
+                "Checking to see if a request is in process"
+            )
+            create_date = fields.Datetime.from_string(authorisation_tracker.create_date)
+            time_10_mins_ago = (datetime.utcnow() - timedelta(minutes=10))
+            if authorisation_tracker:
+                authorised_code_expired = create_date <= time_10_mins_ago
+                if authorised_code_expired:
+                    # we can place a new request and close this request.
+                    authorisation_tracker.closed = 'timed_out'
+                    self._logger.info(
+                        "Connection button Clicked - endpoint name {name}, no Pending requests".format(name=self.name)
+                    )
+                    return self.get_user_authorisation()
+                else:
+                    # The request made was within 10 mins so the user has to wait.
+                    raise exceptions.Warning(
+                        "An authorisation request is already in process!!!\n " +
+                        "Please try again later"
+                    )
+            else:
+                return self.get_user_authorisation()
+
+    def _json_command(self, command, timeout=3, record_id=None):
+        # this will determine where we have come from
+        # if there is no record_id then we know we already had valid record to gain all the information for hello user
+        self._logger.info(
+            "_json_command - has record_id been provided?:- {record_id}".format(record_id=record_id)
+        )
+        if record_id:
+            self = self.env['mtd.hello_world'].search([('id', '=', record_id)])
+            self._logger.info(
+                "_json_command - we need to find the record and assign it to self"
+            )
+        # search the token table to see if we have access token
+        token_record = self.env['mtd.api_tokens'].search([('api_id', '=', self.api_name.id)])
+        access_token = token_record.access_token if token_record else ""
```

Comment:
```
access_token = token_record.access_token or ""
```

If `token_record` is zero-length, then `token_record.access_token` will return `False`.

and

`(False or "") == ""`